docs(validation): explain P-026 and P-027 handling (#309)
camaraproject/tooling: documentation/validation/faq.md

</details>

<a name="p-026-p-027-info-description-mandatory"></a>
<details>
<summary>[P-026]/[P-027] What should I do about missing or drifted mandatory <code>info.description</code> blocks?</summary>

Applies to: `[P-026] Mandatory info.description template missing`, `[P-027] info.description mandatory template content drift`

CAMARA API specifications must include mandatory `info.description` text blocks from the API Design Guide and the Identity and Consent Management profile. Starting with Commonalities r4.3, the canonical text is synchronized into each API repository as `code/common/info-description-templates.yaml`. The API definition marks each mandatory block with `<!-- BEGIN: ... -->` and `<!-- END: ... -->` delimiters, and validation compares the delimited content with the canonical file.

The severity depends on the API status in `release-plan.yaml`:

- `target_api_status: alpha`: P-026/P-027 are hints. The blocks should be added or corrected before the API progresses to rc.

> - `target_api_status: rc`: P-026/P-027 are warnings. The API can still use the r4.3 sync PR to receive the canonical file, but codeowners should fix the warnings before the rc pre-release.

Reviewer: can you clarify what is "the canonical file" ?

Author: changed in https://github.com/camaraproject/tooling/pull/309/commits/b9bea0008105ff4ffcf29bfc6f5b1ab4a130abb8, please check.

> - `target_api_status: public`: P-026/P-027 are errors. The API is expected to contain the mandatory blocks verbatim; validation fails and would block the release until they are fixed.

Reviewer: ```suggestion
- `target_api_status: public`: P-026/P-027 are errors. The API is expected to contain the mandatory blocks verbatim; validation fails and will block the release until they are fixed.
```

Author: applied in https://github.com/camaraproject/tooling/pull/309/commits/b9bea0008105ff4ffcf29bfc6f5b1ab4a130abb8

> **What you do:** copy the reported template from `code/common/info-description-templates.yaml` into the affected API file's `info.description`, keeping the BEGIN/END delimiters and the canonical text unchanged. For P-027, replace the whole drifted delimited block with the matching block from the canonical file.

Reviewer: ```suggestion
**What you do:** on main, copy the reported template from `code/common/info-description-templates.yaml` into the affected API file's `info.description`, keeping the BEGIN/END delimiters and the copied text unchanged. For P-027, replace the whole drifted delimited block with the matching block from the local  `code/common/info-description-templates.yaml` file.
```

> If the finding appears on the automated Commonalities r4.3 sync pull request itself, first check the API status. For alpha or rc APIs, merge the sync PR so the repository receives `code/common/info-description-templates.yaml`, then open a follow-up API PR to copy the canonical blocks into `info.description`. For APIs already marked `public`, change the release plan only if the API is actually still preparing an rc release; otherwise fix the missing or drifted blocks before release.

Reviewer: ```suggestion
If the finding appears when merging the automated Commonalities r4.3 sync PR itself, first check the API status in the `release-plan.yaml`. For alpha or rc APIs, merge the sync PR so the repository receives the required `code/common/info-description-templates.yaml`, then open a follow-up API PR to copy the common mandatory blocks into the API's `info.description`. For APIs already marked `public`, change the release plan only if the API is actually still preparing an rc release; otherwise fix the missing or drifted blocks before release.
```

Author: applied in https://github.com/camaraproject/tooling/pull/309/commits/b9bea0008105ff4ffcf29bfc6f5b1ab4a130abb8

> If the finding appears on the automated Commonalities r4.3 sync pull request itself, first check the API status. For alpha or rc APIs, merge the sync PR so the repository receives `code/common/info-description-templates.yaml`, then open a follow-up API PR to copy the canonical blocks into `info.description`. For APIs already marked `public`, change the release plan only if the API is actually still preparing an rc release; otherwise fix the missing or drifted blocks before release.

Reviewer: the last sentence is not clear to me

Author: updated in https://github.com/camaraproject/tooling/pull/309/commits/b9bea0008105ff4ffcf29bfc6f5b1ab4a130abb8, please check.
